src/rt/array/concat.d

```diff
@@ -0,0 +1,102 @@
+/**
+ This module contains support for controlling dynamic arrays' concatenation
+  Copyright: Copyright Digital Mars 2000 - 2019.
+  License: Distributed under the
+       $(LINK2 http://www.boost.org/LICENSE_1_0.txt, Boost Software License 1.0).
+     (See accompanying file LICENSE)
+  Source: $(DRUNTIMESRC rt/_array/_concat.d)
+*/
+module rt.array.concat;
+
+/// See $(REF _d_arraycatnTX, rt,lifetime,_d_arraycatnTX)
+private extern (C) void[] _d_arraycatnTX(const TypeInfo ti, byte[][] arrs) pure nothrow;
+
+/**
+ * Concatenating the arrays inside of `arrs`.
+ * `_d_arraycatnTX([a, b, c])` means `a ~ b ~ c`.
+ * Params:
+ *  arrs = Array containing arrays that will be concatenated.
+ * Returns:
+ *  A newly allocated array that contains all the elements from all the arrays in `arrs`.
+ * Bugs:
+ *   The safety level of this function is faked. It shows itself as `@trusted pure nothrow` to not break existing code.
+ */
```

Comment:
Maybe also copy the more detailed explanation regarding faked safety from the other PR here.

Reply:
:+1: